### PR TITLE
Weba/update nuphar doc

### PR DESCRIPTION
**Pull request by baowenlei**
**Description**: Update Nuphar Provider Doc.

**Motivation and Context**
- Update Nuphar related configurations.

### CODE REVIEW

File: docs/execution_providers/Nuphar-ExecutionProvider.md [hunk 1]
```diff
@@ -63,6 +63,22 @@ create_shared.sh -c /path/to/jit/cache/NUPHAR_CACHE_VERSION [-m optional_model_f
...
 
+### NGEMM
+NGEMM (Nuphar GEMM) is an optimized low-precision GEMM implementation based on compiler techniques.
+Please refer to our paper for more details of NGEMM: ["NGEMM: Optimizing GEMM for Deep Learning via Compiler-based Techniques"](https://arxiv.org/abs/1910.00178).
+
+#### NGEMM Tiling / Permutation Configuration
+NGEMM has default tiling parameters, but users can overwrite them through environment variables:
+* NUPHAR_IGEMM_TILE_M / NUPHAR_IGEMM_TILE_N / NUPHAR_IGEMM_TILE_K
+
+    These 3 parameters are the tiling sizes for corresponded dimensions of GEMM ([M x K] x [K x N]).
```
Comment (yangchen-MS) [inline docs/execution_providers/Nuphar-ExecutionProvider.md]: typo: corresponded -> the corresponding

File: docs/execution_providers/Nuphar-ExecutionProvider.md [hunk 2]
```diff
@@ -63,6 +63,22 @@ create_shared.sh -c /path/to/jit/cache/NUPHAR_CACHE_VERSION [-m optional_model_f
...
+    Set them to different values will generate GEMM with different tiling sizes.
```
Comment (yangchen-MS) [inline docs/execution_providers/Nuphar-ExecutionProvider.md]: typo: Set -> Setting

File: docs/execution_providers/Nuphar-ExecutionProvider.md [hunk 3]
```diff
@@ -63,6 +63,22 @@ create_shared.sh -c /path/to/jit/cache/NUPHAR_CACHE_VERSION [-m optional_model_f
...
+
+* NUPHAR_IGEMM_PERMUTE
+
+    This enviornment variable is to control the loop permutation in GEMM.
+    Default is not apply any loop permutation. Other options are "inner/outer/all", corresponding to apply permutation for only inner tile loops / only outer loops / both inner and outer loops.
```
Comment (yangchen-MS) [inline docs/execution_providers/Nuphar-ExecutionProvider.md]: typo: "Default is not" -> "The default is to not"

Comment (yangchen-MS) [reply docs/execution_providers/Nuphar-ExecutionProvider.md]: More typo: "corresponding to apply permutation for" -> "to "referring to applying permutations to ..., respectively."

Comment (baowenlei) [reply docs/execution_providers/Nuphar-ExecutionProvider.md]: Thanks.

File: docs/execution_providers/Nuphar-ExecutionProvider.md [hunk 4]
```diff
@@ -63,6 +63,22 @@ create_shared.sh -c /path/to/jit/cache/NUPHAR_CACHE_VERSION [-m optional_model_f
...
 
+### NGEMM
+NGEMM (Nuphar GEMM) is an optimized low-precision GEMM implementation based on compiler techniques.
+Please refer to our paper for more details of NGEMM: ["NGEMM: Optimizing GEMM for Deep Learning via Compiler-based Techniques"](https://arxiv.org/abs/1910.00178).
+
+#### NGEMM Tiling / Permutation Configuration
+NGEMM has default tiling parameters, but users can overwrite them through environment variables:
+* NUPHAR_IGEMM_TILE_M / NUPHAR_IGEMM_TILE_N / NUPHAR_IGEMM_TILE_K
+
+    These 3 parameters are the tiling sizes for corresponding dimensions of GEMM ([M x K] x [K x N]).
```
Comment (yangchen-MS) [inline docs/execution_providers/Nuphar-ExecutionProvider.md]: typo: "corresponding" -> "the corresponding"

File: docs/execution_providers/Nuphar-ExecutionProvider.md [hunk 5]
```diff
@@ -63,6 +63,22 @@ create_shared.sh -c /path/to/jit/cache/NUPHAR_CACHE_VERSION [-m optional_model_f
...
+    Setting them to different values will generate GEMM with different tiling sizes.
+
+* NUPHAR_IGEMM_PERMUTE
+
+    This enviornment variable is to control the loop permutation in GEMM.
+    The default is to not apply any loop permutation. Other options are "inner/outer/all",referring to apply permutation to only inner tile loops / only outer loops / both inner and outer loops, respectively.
```
Comment (yangchen-MS) [inline docs/execution_providers/Nuphar-ExecutionProvider.md]: typo: permutation -> permutations

File: docs/execution_providers/Nuphar-ExecutionProvider.md (hunk 4 above)
Comment (baowenlei) [reply docs/execution_providers/Nuphar-ExecutionProvider.md]: Not a typo. the + plural count nouns are used for specific reference, cannot used for general reference "corresponding dimensions" here.

Comment (yangchen-MS) [reply docs/execution_providers/Nuphar-ExecutionProvider.md]: Not want to be too pedantic, but "of GEMM" requires us to add "the" in front of corresponding. I think we need a native English speaker to help us :)

**Pull request merged**